httpbin/cmd/cmd.go

```diff
@@ -101,7 +107,11 @@ func mainImpl(args []string, getEnv func(string) string, getHostname func() (str
 			MaxHeaderBytes:    srvMaxHeaderBytes,
 			ReadHeaderTimeout: srvReadHeaderTimeout,
 			ReadTimeout:       srvReadTimeout,
-			TLSNextProto:      make(map[string]func(*http.Server, *tls.Conn, http.Handler)),
+		}
+		if cfg.EnableHTTP2 {
+			tlsSrv.Handler = h2c.NewHandler(tlsSrv.Handler, &http2.Server{})
+		} else {
+			tlsSrv.TLSNextProto = make(map[string]func(*http.Server, *tls.Conn, http.Handler))
```

Comment:
is that on purpose that you set different fields of the struct based on the protocol?

Reply:
yes, if http2 is not enabled we need to override the TLSNextProto to disable http2
if it is enabled, we need to wrap the handle